### PR TITLE
feat(app): add odd typography and update StyledText

### PR DESCRIPTION
**Pull request by koji**
# Overview
Add level1header - level4header + body text + small text to typography
update StyledText

# Test Plan

# Changelog
- add new constants to TYPOGRAPHY
- update StyledText with media query for touchscreen
- use updated StyledText for Navigation text (level3header)

# Review requests

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/OnDeviceDisplay/Navigation/index.tsx [hunk 1]
```diff
@@ -64,26 +67,13 @@ export function Navigation({ routes }: { routes: RouteProps[] }): JSX.Element {
...
           {navRoutes.map(({ name, navLinkTo }: RouteProps) => (
             <NavigationLink key={name} to={navLinkTo as string}>
-              <StyledText
-                fontSize={TYPOGRAPHY.fontSize32}
-                fontWeight={TYPOGRAPHY.fontWeightSemiBold}
-                lineHeight={TYPOGRAPHY.lineHeight42}
-                color={COLORS.darkBlack_seventy}
-                marginRight={TYPOGRAPHY.fontSize32}
-              >
+              <StyledText as="h3" marginRight={TYPOGRAPHY.fontSize32}>
```
Comment (ewagoner) [inline app/src/organisms/OnDeviceDisplay/Navigation/index.tsx]: Should this use a value from SPACING, not TYPOGRAPHY (even if the actual value is the same)?

Comment (koji) [reply app/src/organisms/OnDeviceDisplay/Navigation/index.tsx]: Good catch!
Yeah, you are right. That should be SPACING.spacing6.
I will fix.

**Comment by koji** [comment]
- [x] fix test issue

File: app/src/atoms/MenuList/OverflowBtn.tsx
Comment (b-cooper) [inline app/src/atoms/MenuList/OverflowBtn.tsx:1]: The `isOnDevice` variant of this button was only being used in one place, and there was almost nothing shared between the two variants. I've split this out for clarity

File: app/src/atoms/Toast/index.tsx
Comment (b-cooper) [inline app/src/atoms/Toast/index.tsx:1]: Confirmed with Design that the old `fontWeightBold` value of 800 (which was only used in three locations) should be wholesale replaced by the new `fontWeightBold` value of 700 (which had been stored in this clunky variable `fontWeightLevel2_bold`).  From now on, Bold is Bold no matter where the text is being rendered.

File: app/src/organisms/OnDeviceDisplay/Navigation/index.tsx [hunk 2]
```diff
@@ -4,106 +4,116 @@ import { NavLink } from 'react-router-dom'
...
+import { ODD_FOCUS_VISIBLE } from '../../../atoms/buttons/OnDeviceDisplay/constants'
 import { getLocalRobot } from '../../../redux/discovery'
 import { NavigationMenu } from './NavigationMenu'
 
 import type { RouteProps } from '../../../App/types'
 
 export function Navigation({ routes }: { routes: RouteProps[] }): JSX.Element {
   const localRobot = useSelector(getLocalRobot)
-  const [showNavMenu, setNavMenu] = React.useState<boolean>(false)
+  const [showNavMenu, setShowNavMenu] = React.useState<boolean>(false)
```
Comment (b-cooper) [inline app/src/organisms/OnDeviceDisplay/Navigation/index.tsx:29]: Updated the name of this setter function to match the convention elsewhere in this app of  `[foo, setFoo]`

File: app/src/organisms/OnDeviceDisplay/Navigation/index.tsx [hunk 3]
```diff
@@ -4,106 +4,116 @@ import { NavLink } from 'react-router-dom'
...
-  `
   return (
     <>
       <Flex
         flexDirection={DIRECTION_ROW}
         alignItems={ALIGN_CENTER}
         justifyContent={JUSTIFY_SPACE_BETWEEN}
-        paddingTop={SPACING.spacingXXL}
-        marginBottom={SPACING.spacing5}
+        height="124px"
```
Comment (b-cooper) [inline app/src/organisms/OnDeviceDisplay/Navigation/index.tsx:41]: This height value is defined in pixels intentionally and is reinforced by a conversation with Mel in person. The design of this navigation bar is meant to have a fixed height in all places. It's contents are center aligned. and the distance between the items and the top of the viewport is defined by this relationship, not page-level or component level padding/margin

File: app/src/organisms/OnDeviceDisplay/Navigation/index.tsx [hunk 4]
```diff
@@ -4,106 +4,116 @@ import { NavLink } from 'react-router-dom'
...
         />
       )}
     </>
   )
 }
+
+const NavigationLink = (props: { to: string; name: string }): JSX.Element => (
+  <TouchNavLink to={props.to}>
+    {props.name}
+    <Box width="2.5rem" height="5px" borderRadius="2px" />
```
Comment (b-cooper) [inline app/src/organisms/OnDeviceDisplay/Navigation/index.tsx:81]: This implementation of nav link adds the designed border radius to the purple bar.

File: app/src/organisms/OnDeviceDisplay/RobotDashboard/EmptyRecentRun.tsx [hunk 5]
```diff
@@ -21,6 +22,7 @@ export function EmptyRecentRun(): JSX.Element {
       width="100%"
       height="27.25rem"
       backgroundColor={`${COLORS.darkBlackEnabled}${COLORS.opacity15HexCode}`}
+      borderRadius={BORDERS.size_three}
```
Comment (b-cooper) [inline app/src/organisms/OnDeviceDisplay/RobotDashboard/EmptyRecentRun.tsx:25]: border radius added to match design

**Review by b-cooper — APPROVED** [review]
🌸

**Pull request merged**